extensions/vscode/src/views/configurations.ts

```diff
@@ -75,10 +75,22 @@ export class ConfigurationsTreeDataProvider
 
     try {
       const api = await useApi();
-      const response = await api.configurations.getAll({
+      const getAllPromise = api.configurations.getAll({
         dir: ".",
         recursive: true,
       });
+
+      window.withProgress(
+        {
+          title: "Initializing",
+          location: { viewId: Views.Configurations },
+        },
+        async () => {
+          return getAllPromise;
+        },
```

Comment:
Can we just pass the `getAllPromise` here rather than wrapping it in another function?

Reply:
API is expecting a function returned.